src/Microsoft.Azure.WebJobs.Host/Loggers/Logger/AppInsights/AppInsightsLogger.cs

```diff
@@ -0,0 +1,312 @@
+                        break;
+                    case LoggingKeys.StartTime:
+                        DateTimeOffset startTime = new DateTimeOffset((DateTime)prop.Value, TimeSpan.Zero);
+                        requestTelemetry.Timestamp = startTime;
+                        requestTelemetry.Properties.Add(prop.Key, startTime.ToString(DateTimeFormatString));
+                        break;
+                    case LoggingKeys.Duration:
+                        if (prop.Value is TimeSpan)
+                        {
+                            requestTelemetry.Duration = (TimeSpan)prop.Value;
+                        }
+                        break;
+                    case LoggingKeys.OriginalFormat:
+                        // this is the format string; we won't use it here
+                        break;
+                    default:
+                        if (prop.Value is DateTime)
+                        {
+                            DateTimeOffset date = new DateTimeOffset((DateTime)prop.Value, TimeSpan.Zero);
+                            requestTelemetry.Properties.Add(prop.Key, date.ToString(DateTimeFormatString));
+                        }
+                        else
+                        {
+                            requestTelemetry.Properties.Add(prop.Key, prop.Value?.ToString());
+                        }
+                        break;
+                }
+            }
+        }
+
+        private static void ApplyScopeProperties(ITelemetry telemetryModel)
+        {
+            IDictionary<string, object> scopeProps = AppInsightsScope.GetMergedStateDictionary() ?? new Dictionary<string, object>();
+
+            object functionInvocationId;
+            if (scopeProps.TryGetValue(ScopeKeys.FunctionInvocationId, out functionInvocationId))
+            {
+                telemetryModel.Context.Operation.Id = functionInvocationId.ToString();
+            }
+
+            object functionName;
+            if (scopeProps.TryGetValue(ScopeKeys.FunctionName, out functionName))
+            {
+                telemetryModel.Context.Operation.Name = functionName.ToString();
+            }
+        }
+
+        public bool IsEnabled(LogLevel logLevel)
+        {
+            if (_filter == null)
+            {
+                return true;
+            }
+
+            return _filter(_categoryName, logLevel);
+        }
+
+        public IDisposable BeginScope<TState>(TState state)
+        {
+            if (state == null)
+            {
+                throw new ArgumentNullException(nameof(state));
+            }
+
+            return AppInsightsScope.Push(state);
+        }
+
+        internal static string GetIpAddress(HttpRequestMessage httpRequest)
+        {
+            HttpContextBase context = httpRequest.Properties.GetValueOrNull<string, object, HttpContextBase>(ScopeKeys.HttpContext);
+            return context?.Request?.UserHostAddress ?? "0.0.0.0";
```

Comment:
Yes. For Azure Web Apps IP is always in that header. Make sure to cut out the port - if I'm not mistaken it may be an issue with web apps's load balancer